Activity details fixes #149 #151

AlexLakatos opened the pull request:
This is how activities look now.
ToDo in a later PR:
- better icon images for activity detail pages
- localized versions
![webvr camp activate mozilla](https://cloud.githubusercontent.com/assets/690132/23971624/951517b2-09c6-11e7-83ed-e397da3e8796.png)

MichaelKohler commented on _layouts/activity-default.html:
```diff
@@ -0,0 +1,26 @@
+    {% include header.html %}
+
+    <div class="page-content">
+      <div class="wrapper activity-page">
```
from what I remember without checking, the activity-page class is the only thing that changes from the default layout. Could you check if there is any other way how we could inject that so we don't need to do a separate file?

AlexLakatos replied:
Been there, tried stuff. Jekyll has no page information at that level. Couldn't find a better way.

MichaelKohler commented on _pages/community-participation-guidelines.md:
```diff
@@ -1,26 +1,26 @@
-layout: page
+layout: activity
 title:  "Community Participation Guidelines (Draft) Review"
+subtitle: "In like 2-3 hours"
```
can we get rid of the "like" in all activities?

MichaelKohler replied:
also let's make the "In" lowercase

MichaelKohler commented on _pages/community-participation-guidelines.md:
```diff
@@ -1,26 +1,26 @@
 ---
 
+<div class="col-md-3 pull-right" markdown="1">
 * TOC
```
let's get rid of the TOC, and add sections as discussed through Telegram. Also, let's add the "impact and metrics" on the right side (totally fine to not show on mobile)

MichaelKohler commented on _pages/community-participation-guidelines.md:
```diff
@@ -83,10 +83,10 @@ You can sign up for a call using calendly.  Please select one of:
 **OR...**
 
-* Join our #Mozillians Telegram & Twitter live chat on [March 9th](https://www.timeanddate.com/worldclock/fixedtime.html?msg=CPG+Telegram+%26+Twitter+Chat&iso=20170309T07&p1=1091&ah=1) ! 
+* Join our #Mozillians Telegram & Twitter live chat on [March 9th](https://www.timeanddate.com/worldclock/fixedtime.html?msg=CPG+Telegram+%26+Twitter+Chat&iso=20170309T07&p1=1091&ah=1) !
```
as you have several of those fixes, want to add an .editorconfig?

AlexLakatos replied:
don't think I have one, but you can add an issue for it.

MichaelKohler commented on _pages/e10s-addons.md:
```diff
@@ -1,15 +1,21 @@
-layout: page
+layout: activity
 title:  "e10s and Featured Add-ons"
+subtitle: "In like no time because it's retired"
```
I'd remove the subtitle completely for retired ones. Also, can we add a light red background to the text or something similar to better show that this is retired?

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -6,7 +6,6 @@
     min-height: 40px;
     background: $brand-color;
     height: 60px;
-
```
I like to have a linebreak before comments, but I guess that's a matter of taste.

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -42,9 +41,11 @@
 
-    &, &:visited {
+    &,
+    &:visited {
```
thx!

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -331,6 +325,36 @@
     }
 
+    .activity-page {
+      padding: 0 0 20px 0;
```
could you change that whole block to have 4 spaces as you changed above? ;)

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -331,6 +325,36 @@
 
+    .activity-page {
+      padding: 0 0 20px 0;
+      background-color: #DAE3E7;
```
given the fact that we want to adopt to mozilla style colors in the future, could you please add this as a variable?

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -331,6 +325,36 @@
+        .col-md-3 {
+          width: 24%;
+          margin-right: -1%;
+          background-color: #FFFFFF;
```
given the fact that we want to adopt to mozilla style colors in the future, could you please add this as a variable?

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -498,69 +517,58 @@
 .post-header {
     margin-bottom: $spacing-unit;
+    padding: 30px 0;
+    background: #f5f5f5;
```
given the fact that we want to adopt to mozilla style colors in the future, could you please add this as a variable?

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -573,27 +581,28 @@
-
 /* Language list */
-.en:lang(en), .es:lang(es), .pt:lang(pt), .fr:lang(fr) {
+.en:lang(en),
```
can we make this somehow less generic to how many languages we have?

AlexLakatos replied:
I have no idea

Ryuno-Ki replied:
Maybe via CSS variables?

Check browser support, but it could be an enhancement.

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -676,22 +718,6 @@ div.facebook {
     margin-top: -20px;
 }
 
-#markdown-toc {
```
This breaks the FAQ TOC, but I kind of like it better like that.. ;)

![screen shot 2017-03-15 at 23 18 39](https://cloud.githubusercontent.com/assets/330324/23973290/c782acaa-09d5-11e7-9306-75489ac8554c.png)

AlexLakatos replied:
I know, me too 🐶

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -751,28 +778,4 @@ div.facebook {
 }
-
 // Bootstrap Overwrites
-table.sortable > thead th:hover:not([data-defaultsort="disabled"]) {
```
why is this block removed?

* This breaks the hover on the tables and we again have an unreadable color
* This breaks the menu on md devices

AlexLakatos replied:
messed up the merge conflict I guess

MichaelKohler commented:
Now, let's talk responsive.

* It looks like the TOC is not hidden on smaller screens
![screen shot 2017-03-15 at 23 28 20](https://cloud.githubusercontent.com/assets/330324/23973603/2104ee22-09d7-11e7-9b1a-83618a96c02f.png)
* I would remove the 1% border around the white background on mobile
* The picture in the header might need some margin on the left (or even break it to its own line).. Feel free to use flexbox
![screen shot 2017-03-15 at 23 30 25](https://cloud.githubusercontent.com/assets/330324/23973657/6948d9dc-09d7-11e7-993f-a2edbf9315f6.png)
* Let's add a ```text-align: justify;``` for paragraphs on mobile. I think the following would look better:
![screen shot 2017-03-15 at 23 32 43](https://cloud.githubusercontent.com/assets/330324/23973727/ba26ec18-09d7-11e7-8ee1-86f7d30d3002.png)
* The first few paragraphs get earlier line-breaks than the ones after them, this might be because of the TOC though
![screen shot 2017-03-15 at 23 33 11](https://cloud.githubusercontent.com/assets/330324/23973749/d6631528-09d7-11e7-86dd-c32d27bac3e5.png)

AlexLakatos commented:
Addressed most of the review.
I have no idea what images to use, maybe @couci can help me here? I need 150x150 pixel images please.
I think localization should happen separately, this PR is getting big enough already.

![webvr camp activate mozilla 1](https://cloud.githubusercontent.com/assets/690132/23979534/040f66f6-09f2-11e7-837b-eca2babc8ec2.png)

MichaelKohler commented on _sass/_layout.scss:
```diff
@@ -751,28 +790,8 @@ div.facebook {
-  background-color: $brand-color-light;
-}
-
-@include media-query($on-laptop) {
```
this here should be in the PR as well (see comment above). Otherwise the menu looks really weird on md devices

MichaelKohler commented:
Thanks! This is on the right track. I think we need to fix the images before we merge. The rest is just some smaller nits:

* Looks like the border is still there at the left side of the boxes on mobile
<img width="105" alt="screen shot 2017-03-16 at 21 26 53" src="https://cloud.githubusercontent.com/assets/330324/24017212/5d71d786-0a8f-11e7-9308-833163aae5c0.png">
* I can scroll a few pixels to the right for everything below 821px width
* Let's center the title of the activity as well as the social buttons on mobile
<img width="475" alt="screen shot 2017-03-16 at 21 28 34" src="https://cloud.githubusercontent.com/assets/330324/24017262/8f94b4ae-0a8f-11e7-9597-68c7a42bda8b.png">
* See comment about md menu

MichaelKohler commented:
P1: merge of this
P2: file a follow up issue for l10n work

couci commented:
@AlexLakatos I am (finally) gathering your pictures. Could you indicate which ones need an alternative picture from what we have on the launch page?

couci commented:
![foxface_puzzle-600x585](https://cloud.githubusercontent.com/assets/7702955/24856589/f08d0eb6-1ded-11e7-9c9d-3184343cdb1c.jpeg)
Addons added

couci commented:
A-frame ones here: https://drive.google.com/drive/u/0/folders/0B1WwYOKD6M7vWWFLOWpiclhGaVU (thanks to @Srushtika )

couci commented:
<img width="230" alt="screen shot 2017-04-10 at 13 23 57" src="https://cloud.githubusercontent.com/assets/7702955/24857458/1aab3102-1df1-11e7-8c5b-16d91f4b3d6a.png">
Webcompatability

couci commented:
For test pilot (retired) let's use the firefox one same for e-10s. As for the CPG let's use the m from mozilla new logo since there is not a specific brand for there
https://assets.mozilla.org/portal/assets/#asset/205

MichaelKohler commented:
I've updated the layout for ES/PT as well. Also I've deleted FR completely, as there is no need to keep this up to date with EN as long as there are no translations.

r+ from me for all changes, this can be merged if Laka agrees with my changes and has done the image swaps.
